### DISCUSSION TITLE
Prevent deletion of /etc/resolv.conf on session start

### DISCUSSION THREAD

**Discussion by avinashupadhya99** [WhitewaterFoundry/Pengwin#705] [Q&A] [answered]
Hi, I had issues with DNS and the solution that worked for me was running `echo nameserver 1.1.1.1 > /etc/resolv.conf`. I also disabled `generateResolvConf` in `/etc/wsl.conf` by setting it to false. But everytime I open a new terminal window, the `/etc/resolv.conf` is deleted and hence I'm not able to access the internet. Any permanent fix for this? Thanks

Once the file was deleted, not able to edit or create a new one as well -
```
root@LAPTOP-FMG2BKC5:~# echo nameserver 1.1.1.1 > /etc/resolv.conf
bash: /etc/resolv.conf: No such file or directory
root@LAPTOP-FMG2BKC5:~# touch /etc/resolv.conf
touch: cannot touch '/etc/resolv.conf': No such file or directory
root@LAPTOP-FMG2BKC5:~# cat /etc/resolv.conf
cat: /etc/resolv.conf: No such file or directory
```

**Comment by avinashupadhya99** [comment, marked as answer]
Update: Following this seemed to have fixed the issue - https://colhountech.com/2021/04/01/wsl2-ubuntu-ping-hostname-temporary-failure-in-name-resolution/ The file was not deleted but rather the symbolic link was broken

**Comment by crramirez** [comment]
Hello @avinashupadhya99,

generateResolvConf should fix the problem. I'll take a look. To reproduce it and how was fixed with the article that you linked.

Regards,
Carlos

**Reply by crramirez** [reply]
I see that the article that you posted is the solution for the problem

**Reply by avinashupadhya99** [reply]
Setting `generateResolvConf ` to false did not prevent the breaking of the symbolic link. So the internet access was cut off even after setting `generateResolvConf ` to false. But the article helped later.
